storage/mysql/log_storage.go

```diff
@@ -348,9 +349,10 @@ func (m *mySQLLogStorage) QueueLeaves(ctx context.Context, tree *trillian.Tree,
 
 type logTreeTX struct {
 	treeTX
-	ls   *mySQLLogStorage
-	root types.LogRootV1
-	slr  *trillian.SignedLogRoot
+	ls       *mySQLLogStorage
+	root     types.LogRootV1
+	slr      *trillian.SignedLogRoot
+	dequeued map[string]dequeuedLeaf
```

Comment:
Accumulating leaves in the tx seems a big strange. Is this the only option that works?

Reply:
Following the spanner impl... 

I think the API is a bit strange, but at least it's consistent.  

One we have good test coverage across everything, we could consider changing the API.